### ISSUE TITLE
Allow setting an origin

### ISSUE THREAD

**Issue by stephenplusplus** [stephenplusplus/gcs-resumable-upload#6]
@janesconference - any interest in adding support for any options we're currently missing?

**Comment by janesconference** [comment]
Hi @stephenplusplus, you're right, I should have opened a PR, but I'm totally swamped right now. What I have is a POC that works simply setting

`  this.origin = cfg.origin`

in the `function Upload` constructor

Then, in `Upload.prototype.createURI`, I do:

```
if (this.origin) {
    reqOpts.headers = {
      'Origin': this.origin
    }
  }
```

Then, from my server, I crudely instantiate:

```
createURI({
      authClient: file.bucket.storage.authClient,
      bucket: file.bucket.name,
      file: file.name,
      generation: file.generation,
      metadata: {},
      origin: origin
    }, function(err, url) {
      if (err) {
        cb(err, null)
        return
      }

      cb(null, {
        url: url,
        id: fileObj.id
      })
    })
```

where file is a valid file object created via  bucket.file.

When a browser client calls my node API, I just get the client's origin with `req.get('origin')` (optionally checking it) and everything works.

I wish I had time to put it together in a nice PR, but for now I don't even have that one hour to dedicate to it :/

**Comment by stephenplusplus** [comment]
No problem at all! Thanks for sending this.

**Comment by stephenplusplus** [comment]
Released in 0.4.0: https://github.com/stephenplusplus/gcs-resumable-upload/commit/8f961676fd7c3a7afef8a9707c8b7fbb9c15138e
